src/arweave/api.ts

```diff
@@ -0,0 +1,50 @@
+import Axios from 'axios';
+import { estimateArweaveResponse, uploadArweaveResponse } from '../types';
+import { LIKE_CO_API } from '../constant';
+import { createNewForm } from './form';
+import isNode from './util';
+
+const axios = Axios.create({
+  baseURL: LIKE_CO_API,
+  timeout: 60000,
+});
+
+export async function estimateArweavePrice(files: string | File[] | FileList)
+: Promise<estimateArweaveResponse> {
+  let form: any;
+  if (typeof files === 'string') {
+    if (!isNode()) throw new Error('file path is only supported in node.js');
+    form = await createNewForm(files);
+  } else {
+    const fileArray = Array.from(files);
+    form = await createNewForm();
+    fileArray.forEach((f) => {
+      form.append(f.name, f, f.name);
+    });
+  }
```

Comment:
Maybe could refactor this part to a new function for the use of `estimateArweavePrice()` and `uploadToArweave()`?

Reply:
Indeed, but lets invoke rule of three and refactor later

Reply:
OK!